core/includes/form.inc

```diff
@@ -3692,23 +3710,29 @@ function theme_details($variables) {
   // Special handling for form elements.
   if (isset($element['#array_parents'])) {
     // Assign an HTML ID.
-    if (!array_key_exists('id', $element['#attributes'])) {
-      $element['#attributes']['id'] = $element['#id'];
-    }
+    element_set_attributes($element, array('id'));
```

Comment:
`element_set_attributes()` already checks for `!isset($element['#attributes'][$attribute]` - no need to be double-checking here.